src/agent.ts

```diff
@@ -174,12 +174,21 @@ export default class SocksProxyAgent extends Agent {
 			// this socket connection to a TLS connection.
 			debug('Upgrading socket connection to TLS');
 			const servername = opts.servername || opts.host;
-			return tls.connect({
+			const tlsSocket = tls.connect({
```

Comment:
If we are going to merge the fix to the public library I think we have to handle the error instead of just hiding. Here is the solution, but I didn't test it:
```ts
return new Promise((resolve, reject) => {
  const tlsSocket = tls.connect({
    ...omit(opts, 'host', 'hostname', 'path', 'port'),
    socket,
    servername,
    ...this.tlsConnectionOptions,
  });
  const onError = (e) => {
    socket.removeAllListeners();
    tlsSocket.removeAllListeners();
    socket.destroy();
    tlsSocket.destroy();
    reject(e);  
  };
  tlsSocket.once('secureConnect', () => {
    tls.removeListener('error', onError);
    resolve(tlsSocket);
  });
  tlsSocket.once('error', onError);
});
```

Reply:
It would be great to test it before merging.